fix: automatically append taxes if taxes_and_charges is set in Buying controller

ljain112 opened the pull request:
Issue: While creating a supplier quotation from a request for quotation, taxes are not populating, even though the taxes and charges template is set.

Steps to replicate:
- Create a tax rule for the supplier.
- Create a Request for Quotation for the supplier.
- Create Supplier quotation from RFQ.

The taxes and charges template will be set, but taxes will be empty.

closes: https://github.com/frappe/erpnext/issues/49610

<!--

Some key notes before you open a PR:

 1. Select which branch should this PR be merged in?
 2. PR name follows [convention](http://karma-runner.github.io/4.0/dev/git-commit-msg.html)
 3. All tests pass locally, UI and Unit tests
 4. All business logic and validations must be on the server-side
 5. Update necessary Documentation
 6. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes


Also, if you're new here

- Documentation Guidelines => https://github.com/frappe/erpnext/wiki/Updating-Documentation

- Contribution Guide => https://github.com/frappe/erpnext/blob/develop/.github/CONTRIBUTING.md

- Pull Request Checklist => https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

-->

> Please provide enough information so that others can review your pull request:

<!-- You can skip this if you're fixing a typo or updating existing documentation -->

> Explain the **details** for making this change. What existing problem does the pull request solve?

<!-- Example: When "Adding a function to do X", explain why it is necessary to have a way to do X. -->

> Screenshots/GIFs

<!-- Add images/recordings to better visualize the change: expected/current behviour -->

khushi8112 committed:
refactor: renamed gross purchase amount to net purchase amount

khushi8112 committed:
refactor: updated gross_purchase_amount to net_purchase_amount across codebase

khushi8112 committed:
chore: added patch to update existing records

khushi8112 committed:
refactor: use correct field name

ravibharathi656 committed:
fix(production plan): filter sales orders by item

elshafei-developer committed:
feat: add employee name to session user

khushi8112 committed:
feat: default print format for sales invoice

khushi8112 committed:
feat: add css

khushi8112 committed:
style: change padding

khushi8112 committed:
feat: add letterhead fixture

khushi8112 committed:
fix: css changes in letterhead

khushi8112 committed:
fix: letterhead styling

khushi8112 committed:
fix: broken img tag in letterhead

khushi8112 committed:
fix: condition based address display

khushi8112 committed:
fix: remove border if company logo not available

khushi8112 committed:
feat: print format design two

khushi8112 committed:
fix: radius of the items/tax table thead

khushi8112 committed:
test: just debugging

khushi8112 committed:
test: add in_install condition for debugging

khushi8112 committed:
refactor: remove img tag for testing

khushi8112 committed:
refactor: revert debugging changes

khushi8112 committed:
style: center-align logo within its container div

khushi8112 committed:
style: always show border even when logo is missing

khushi8112 committed:
fix: do not make letterhead default

KerollesFathy committed:
fix(accounts): update payment mode account retrieval to use namespaced function

khushi8112 committed:
fix: update styles to work with wkhtmltopdf rendering

khushi8112 committed:
refactor: remove flex usage for better wkhtmltopdf support

khushi8112 committed:
refactor: update letterhead styles for wkhtmltopdf compatibility

khushi8112 committed:
refactor: small changes for better readability

khushi8112 committed:
refactor: remove tax breakup table from the print format

khushi8112 committed:
feat: letterhead for print format

diptanilsaha committed:
feat: consolidated trial balance report

diptanilsaha committed:
feat: gl entries with values in reporting_currency

khushi8112 committed:
feat: add default letterhead with HTML template via after_install

khushi8112 committed:
fix: app path correctly

khushi8112 committed:
refactor: create_letter_head for readability

ruthra-kumar committed:
feat: process period closing voucher

khushi8112 committed:
feat: prompt user to input company logo and address if missing in print preview

khushi8112 committed:
fix: better sub total section with tax breakup

starkshade commented on erpnext/controllers/buying_controller.py:
```diff
@@ -216,6 +217,12 @@ def set_missing_values(self, for_validate=False):
 
 		self.set_missing_item_details(for_validate)
 
+		if self.meta.get_field("taxes"):
```
## ⚠️ Code Issue

**Lines:** 220-224

```python
		if self.meta.get_field("taxes"):
			if self.get("taxes_and_charges") and not self.get("taxes") and not for_validate:
				taxes = get_taxes_and_charges("Purchase Taxes and Charges Template", self.taxes_and_charges)
				for tax in taxes:
					self.append("taxes", tax)
```

This code automatically populates taxes based on `taxes_and_charges` within the `set_missing_values` method. While it avoids populating taxes during validation (`for_validate=True`), it might implicitly add taxes in other scenarios where this method is called, which could be unexpected. It's generally better to have explicit control over when such data is populated.

starkshade reviewed (commented):
The changes introduce functionality to automatically populate 'taxes' based on a selected 'taxes_and_charges' template in the buying controller. This aims to streamline the process of applying taxes to purchase documents.

**Key Changes:**
- Imported `get_taxes_and_charges` from `erpnext.controllers.accounts_controller`.
- Added logic in `set_missing_values` to fetch and append taxes from a 'Purchase Taxes and Charges Template' if 'taxes_and_charges' is set but 'taxes' is not, and not in validation mode.

diptanilsaha committed:
test: consolidated trial balance

khushi8112 committed:
fix: show tax breakup in print format

khushi8112 committed:
style: fix layout issues with extended data

khushi8112 committed:
style: format and display the address for improved visual clarity

murtaza-ghadiali committed:
refactor: use naming series configuration for Sales Partner ID

Replaced hardcoded ID assignment with Naming Series configuration so that Sales Partner IDs can be managed via Setup > Naming Series. Fixes #49623

khushi8112 committed:
style: add company and customer name on bill_to and bill_from section

khushi8112 committed:
feat: input website, email, phone_no if not already set in company

ruthra-kumar committed:
refactor: checkbox for pcv controller

khushi8112 committed:
fix: small ui changes

khushi8112 committed:
chore: remove frappe.db.commit

Jaswanth-Sriram-Veturi committed:
perf: avoid unnecessary set_dynamic_labels updates

ruthra-kumar committed:
refactor: more data structure changes

ljain112 commented on erpnext/buying/doctype/request_for_quotation/test_request_for_quotation.py:
```diff
@@ -76,6 +76,46 @@ def test_make_supplier_quotation(self):
+	def test_make_supplier_quotation_with_taxes(self):
+		"""Test automatic tax addition when supplier quotation is created from RFQ taxes_and_charges are set"""
+
+		# Create a Purchase Taxes and Charges Template for testing
+		tax_template = frappe.new_doc("Purchase Taxes and Charges Template")
+		tax_template.doctype = "Purchase Taxes and Charges Template"
+		tax_template.title = "_Test Purchase Taxes Template for RFQ"
+		tax_template.company = "_Test Company"
+		tax_template.append(
+			"taxes",
+			{
+				"charge_type": "On Net Total",
+				"account_head": "_Test Account Service Tax - _TC",
+				"description": "VAT",
+				"rate": 10,
+			},
+		)
+		tax_template.save()
+
+		rfq = make_request_for_quotation()
+		supplier = rfq.get("suppliers")[0].supplier
+
+		tax_rule = frappe.new_doc("Tax Rule")
+		tax_rule.company = "_Test Company"
+		tax_rule.tax_type = "Purchase"
+		tax_rule.Supplier = supplier
+		tax_rule.purchase_tax_template = tax_template.name
+		tax_rule.save()
+
+		sq = make_supplier_quotation_from_rfq(rfq.name, for_supplier=supplier)
+
+		# Verify that taxes_and_charges is set from get_party_details
+		self.assertEqual(sq.taxes_and_charges, tax_template.name)
+
+		# Verify that taxes are automatically added
+		self.assertGreaterEqual(len(sq.get("taxes")), 1)
+
+		tax_rule.delete()
+		tax_template.delete()
+
```
if you check Material Request inherits the BuyingController and there is no such method in material request.

ruthra-kumar committed:
refactor: child table in process pcv

rehanrehman389 committed:
fix: prevent empty Create dropdown when In Process

diptanilsaha committed:
fix(process statement of accounts): allow renaming

ljain112 commented:
@ruthra-kumar

rohitwaghchaure committed:
refactor: old serial nos filter

diptanilsaha committed:
feat(process statement of accounts): added more frequency options for auto email

ravibharathi656 committed:
fix: preserve address if present

ljain112 committed:
perf: optimize validate_qty method to eliminate N+1 query problem

rohitwaghchaure committed:
fix: reserved serial / batch not picked in stock entry

HenningWendtland committed:
feat: add company links to Email Account and Communication (#49721)

Co-authored-by: barredterra <14891507+barredterra@users.noreply.github.com>

diptanilsaha committed:
fix(process statement of accounts): naming of reports

khushi8112 committed:
fix: validate email address

ravibharathi656 committed:
fix(asset movement): clear custodian if not present

ruthra-kumar committed:
refactor: temporarily save balances in JSON

ruthra-kumar committed:
refactor: barebones functions

elshafei-developer committed:
feat: Cache employee name in session data on boot

kavin-114 committed:
fix: sum quantity instead of requried BOM quantity

rohitwaghchaure committed:
fix: incorrect field valuation_rate

rohitwaghchaure committed:
fix: batch qty for expired batches

rohitwaghchaure committed:
fix: incorrect qty in stock levels

rohitwaghchaure committed:
fix: Reset Raw Materials Table button not working

rohitwaghchaure committed:
fix: sales return for product bundle items

elshafei-developer committed:
feat: cache employee name in session data on boot

sokumon committed:
fix: format workstation link correctly

ravibharathi656 committed:
fix: fixed asset register showing opening entries

rohitwaghchaure committed:
test: test case for sales return for product bundle

Commit:
fix: call onload of buying controller in purchase_receipt.js

khushi8112 committed:
style: slight spacing and alignment fix

khushi8112 committed:
feat: condition based item code column

ljain112 committed:
refactor: replace SQL query with Query Builder in fetch_items_with_pending_qty method

rohitwaghchaure committed:
fix: consider negative qty in batch qty calculation

Commit:
fix: fix: incorrect PR status when using set landed cost based on PI rate

rehanrehman389 committed:
feat: add asset name to Asset Depreciations and Balances report

ruthra-kumar committed:
refactor: stable start, pause, resume and completion stages

ankush committed:
perf: Only check transaction deletion record once during req/job

rohitwaghchaure committed:
fix: posting date in serial no

l0gesh29 committed:
fix: hide sales invoice creation for fully returned delivery notes

rohitwaghchaure committed:
feat: service expense account in the company

khushi8112 committed:
refactor: change print format type html to custom

manikandan-s-18 committed:
refactor: add supplier filter in buying

khushi8112 committed:
refactor: replace get_doc with get_value

venkat102 committed:
fix(stock-reconciliation): include inventory dimensions in duplicate validation

vorasmit commented:
Inspiration:
https://github.com/frappe/erpnext/blob/aaca906a0f63fc219ed24ac1e1be92a83c5d46d4/erpnext/controllers/selling_controller.py#L147-L150

khushi8112 committed:
refactor: use get_value to improve performance

khushi8112 committed:
refactor: use query builder to set company address

ravibharathi656 committed:
fix(deferred revenue): validate service stop date

rohitwaghchaure committed:
fix: stock ledger adjustment entry

thomasantony12 committed:
fix: Batch ordering based on the method mentioned in settings

khushi8112 committed:
fix: add filter query for address field

rohitwaghchaure committed:
fix: reset raw materials considering not available batches

thomasantony12 committed:
chore: use get_single_value instead of get_cached_doc

rehanrehman389 committed:
fix: filter sales team to show only active individual salespersons

khushi8112 committed:
refactor: add permission check and minor fixes

frappe-pr-bot committed:
chore: update POT file (#50030)

frappe-pr-bot committed:
fix: sync translations from crowdin (#49959)

kavin-114 committed:
fix: handle multi uom in reserved qty for production plan

Bhavan23 committed:
fix(accounts-receivable): ensure report data with party account currency

diptanilsaha committed:
fix: set default roles on role_profile during reinstallation

Bhavan23 committed:
test(accounts-receivable): add test case to validate report data with party account currency

l0gesh29 committed:
fix: add GROUP BY for dn_detail and convert SQL query to QB

ruthra-kumar committed:
refactor: store closing balance as JSON

khushi8112 committed:
perf: add composite indexes to Advance Payment Ledger Entry table

rohitwaghchaure committed:
fix: delivered qty in reservation entry

khushi8112 committed:
fix: revert unrelated manual modified timestamp change

vorasmit committed:
fix: enhance sub-assembly item handling in raw material request calculations

rohitwaghchaure committed:
fix: show qty in popup while making additional transfer entry

Commit:
refactor: move value inline

akhilakr113 committed:
fix: extend quotation filters to exclude 'Ordered' quotations in 'Get Items From' on Sales Order (#50029)

* feat: show item name in update items dialog for sales and purchase order

* feat: remove the ordered quotation from listing the quotation in sales order get items from

* chore: remove this pr from this

elshafei-developer committed:
fix(Supplier Quotation Comparison): add a missing translate function (#49497)

* Update supplier_quotation_comparison.py

* refactor: text cleaning

matteoarosti committed:
fix: warehouse source reference in production report

rohitwaghchaure committed:
fix: do reposting of first transfer entry based on item-wh combination

rohitwaghchaure committed:
fix: duplicate serial nos

rohitwaghchaure committed:
fix: performance issue by adding index

kavin-114 committed:
fix: swap warehouse labels for return entry

venkat102 committed:
fix(stock-entry): fetch empty batch for finished item

kavin-114 committed:
fix: handle flt conversion for prev_ordered_qty

mihir-kandoi committed:
chore: replace broken links with correct ones

rohitwaghchaure committed:
perf: optimize sql query

mihir-kandoi committed:
feat: subcontracting inward (#47728)

* feat: subcontracting inward

* feat: stock reservation

* feat: subcontracting delivery

* feat: all remaining stuff

* fix: linter errors

* fix: patch

* fix: modify stock entry type validation

* fix: customer provided item cost field mandatory validation

* fix: failing tests

* fix: failing tests

* fix: subcontracting controlller

* refactor: semi final

* refactor: final

* chore: resolve conflicts

* refactor: changes requested

* fix: reservation transfer of extra qty

* fix: consider add cost for customer provided rate field

* test: create test data

* test: subcontracted sales order (partial)

* test: fin

* fix: do not add self RM in DN created from SI

* fix: failing test case

* fix: conflicting function name

* refactor: final changes

* fix: more bugs

* perf: various and major performance improvements

* fix: consider warehouse as well in all queries

* fix: same item code with diff warehouse in manufacture entry

* refactor: readability

* fix: frontend validations

* perf: replace query inside loop with single query

* fix: set additional item flag to true when extra customer provided item is received

* fix: bugs found by coderabbit

* fix: more coderabbit bugs

* fix: add validation to disallow cancellation of manufacturing entry

* perf: use cached values wherever it makes sense

* test: fix redundant insert to child tables

* fix: consider SI return of billed self RM

* fix: bug found by coderabbit

---------

Co-authored-by: Mihir Kandoi <mihirkandoi@Mihirs-MacBook-Air.local>

mihir-kandoi committed:
fix: restore customer field

mihir-kandoi committed:
fix: add delivery warehouse filter

rohitwaghchaure committed:
fix: negative error not throw for backdated entry

ruthra-kumar committed:
refactor: store daily balances based on dimensions key

dimensions key is manually converted to string

kavin-114 committed:
fix: preview stock ledger for manual serial and batch values

akhilnarang committed:
fix: sanitize projects field in tasks webform (#50089)

Signed-off-by: Akhil Narang <me@akhilnarang.dev>

mihir-kandoi committed:
fix: source warehouse validation in manufacturing entry during subcontracting inward

mihir-kandoi committed:
fix(stock): remove filter from query

rehanrehman389 committed:
fix: fiscal year overlap validation for company-specific years

yash14023 committed:
perf: optimize company monthly sales query using date range (#48942)

* perf: optimize company monthly sales query using date range instead of DATE_FORMAT

* perf: optimize company monthly sales query using date range

ruthra-kumar committed:
refactor: build and post gl entries

rohitwaghchaure committed:
fix: adjustment entry

diptanilsaha committed:
fix: added exception handling on service level agreement apply hook (#50096)

* fix: added exception handling on service level agreement apply hook

* Revert "fix: added exception handling on service level agreement apply hook"

This reverts commit dae93aa96f4756cbca4bee8b19e8cd4390617e9b.

* fix: Ignore missing SLA table during install/uninstall

---------

Co-authored-by: Ankush Menat <ankush@frappe.io>

rehanrehman389 committed:
test: fiscal year overlap validation for company-specific years

rohitwaghchaure committed:
fix: validation for negative batch

rehansari26 committed:
test: replace get_doc with new_doc in fiscal year tests

ruthra-kumar committed:
refactor: store results as is and convert at the end

PatrickDEissler committed:
fix(Task): make Timesheet-dependent fields no_copy

ruthra-kumar committed:
refactor: for better readability

ruthra-kumar committed:
refactor: process on submit

l0gesh29 committed:
fix: recalculate amount based on allocated amount

ruthra-kumar committed:
refactor: more stable pause and resume

diptanilsaha committed:
fix(point-of-sale): render payment methods only payment component is visible

ruthra-kumar committed:
refactor: maintain report type on each date

ruthra-kumar committed:
refactor: balances for both P&L and Balance sheet accounts

ruthra-kumar committed:
chore: rename closing balance field

ruthra-kumar committed:
refactor: populate opening balances calculation table

ruthra-kumar committed:
refactor: calculate both balances from single queue

ruthra-kumar committed:
refactor: store closing balance for Balnace sheet accounts

ruthra-kumar committed:
refactor: make Accounts Closing Balance as well

ruthra-kumar committed:
refactor: cleanup and for better readability

Abdeali099 committed:
chore: remove print statement

rohitwaghchaure committed:
fix: internal transfer entry with serial/batch

ruthra-kumar committed:
refactor: utility to convert tuple key to str

ruthra-kumar committed:
refactor: utility to consolidate results from all dates

ruthra-kumar committed:
refactor: smaller methods

barredterra committed:
refactor: simplify expression

This expression always evaluates to avalue that is equal to account_currency.

frappe-pr-bot committed:
chore: update POT file (#50172)

frappe-pr-bot committed:
fix: sync translations from crowdin (#50049)

ruthra-kumar committed:
refactor: more changes

1. 'Accounts Manager' has access to submit, cancel and delete
2. cancel and delete operation of PCV is linked with Proces PCV

ruthra-kumar committed:
refactor: abort processing of all tasks upon cancellation

ruthra-kumar committed:
chore: remove scaffolding

ruthra-kumar committed:
chore: progress bar

ruthra-kumar committed:
refactor: enable legacy controller by default for pcv

frappe-pr-bot committed:
fix: sync translations from crowdin (#50176)

deepeshgarg007 committed:
perf: Add index for faster queries (#50175)

AhmedAbokhatwa committed:
fix: correct monthly sales history (#50056)

* fix: correct monthly sales history

* fix: correct monthly sales history calculation

* chore: remove unrelated file accidentally committed

---------

Co-authored-by: AhmedAbukhatwa <Ahmedabukhatwa1@gmail.com>

frappe-pr-bot committed:
fix: sync translations from crowdin (#50185)

mihir-kandoi committed:
fix: feat: multiple changes related to subcontracting inward

rohitwaghchaure committed:
feat: enable item wise inventory account

rohitwaghchaure committed:
fix: sabb missed in the incorrect serial no valuation report

KerollesFathy committed:
Merge pull request #50153 from KerollesFathy/fix-address-error-when-create-transporter

fix: Address title error when create new transporter on driver

rohitwaghchaure committed:
fix: stock difference value for adjustment entry

deepeshgarg007 committed:
feat: Add posting date param for reverse GL entries

deepeshgarg007 committed:
chore: Linting issues

khushi8112 committed:
fix: set status to Draft for auto-created assets from Purchase Receipt

barredterra committed:
fix(Bank Transaction): make transaction ID non-unique

rohitwaghchaure committed:
fix: get valuation rate based of previous SLEs for material receipt

rohitwaghchaure committed:
test: test cases for item wise inventory account

ravibharathi656 committed:
fix(gross profit): remove customer name from columns

kavin-114 committed:
fix: Pass uom field name to update existing item qty

ravibharathi656 committed:
fix(asset depreciations and balances): showing opening entries

frappe-pr-bot committed:
fix: sync translations from crowdin (#50201)

frappe-pr-bot committed:
chore: update POT file (#50225)

frappe-pr-bot committed:
fix: sync translations from crowdin (#50228)

ljain112 commented:
@ruthra-kumar 

Can you review ?

diptanilsaha committed:
fix(consolidated trial balance): index out of range error

rohitwaghchaure committed:
fix: optimized the slow query to get the batchwise available qty

Bhavan23 committed:
fix(journal-entry): allow copy account currency when duplicating JV

ravibharathi656 committed:
fix: avoid group columns mutation

mihir-kandoi committed:
fix: missed refactoring code

rohitwaghchaure committed:
fix: provision to find and fix incorrect serial and batch bundles

barredterra committed:
refactor: error message in budget

improve translatability

barredterra committed:
fix: placeholder number

khushi8112 committed:
fix: create GL entries via hooks only for submitted assets

rohitwaghchaure committed:
fix: allow bulk edit for bill of material items

frappe-pr-bot committed:
fix: sync translations from crowdin (#50242)

venkat102 committed:
fix: set valuation rate for rejected serial/batch item

venkat102 committed:
test: add unit test for valuation rate on rejected serial/batch item

frappe-pr-bot committed:
fix: sync translations from crowdin (#50269)

rohitwaghchaure committed:
fix: added MPS, MRP report in the manufacturing workspace

rohitwaghchaure committed:
fix: only show inventory accounts in the dropdown

rohitwaghchaure committed:
fix: Serial No and Batch Traceability report

mihir-kandoi committed:
Revert "refactor: add supplier filter in buying"

l0gesh29 committed:
feat: modify accounting dimension as multiselect field

diptanilsaha committed:
fix: added validation for default accounts on company

rohitwaghchaure committed:
fix: stock reservation validation for stock reconciliation

rohitwaghchaure committed:
fix: pause button not working for sub-operation

rohitwaghchaure committed:
fix: hourly rate not fetched on selction of workstation type

l0gesh29 committed:
feat(reports): preserve accounting dimension filters while navigating between reports

rohitwaghchaure committed:
fix: job card timer for mobile view

kavin-114 committed:
fix: handle None in last_valuation_rate check

frappe-pr-bot committed:
chore: update POT file (#50304)

diptanilsaha committed:
fix(accounts): populate correct fields on GL Entry during discount accounting

NihalRoshanCK committed:
refactor: remove company filter

'Bank' doctype has no 'company' field.

KerollesFathy committed:
fix: Respect allowed zero qty in SO and PO based on Buying/Selling settings when update items (#49673)

* fix: respect allowed zero qty in SO and PO based on buying/selling settings

* refactor: remove parent_doctype param due to it's already avaliable

* refactor: rename is_allowed_zero_qty_for to is_allowed_zero_qty

* fix: prevent rate change for unit price items

* fix: unboundlocal variable rate_unchanged

* refactor: only throw on 'zero' qty items

---------

Co-authored-by: ruthra kumar <ruthra@erpnext.com>

rohitwaghchaure committed:
fix: stock reservation cancellation for transfer case

mihir-kandoi committed:
revert: remove transporter filter from buying doctypes

ruthra-kumar committed:
refactor: minor changes on status

1. set to 'In Progress' on start of both legacy and new controller
2. force delete to avoid permission issues
3. default to 1hr timeout

rohitwaghchaure committed:
fix: multiple MRP issues

mihir-kandoi committed:
fix: minor issues in subcontracting inward

rohitwaghchaure committed:
fix: negative stock validation

mihir-kandoi committed:
fix: disallow material transfer if source and target warehouse are same (#48697)

* fix: disallow material transfer if source and target warehouse are same

* fix: check on child row level instead of parent level

mihir-kandoi committed:
chore: add docstring to function

PugazhendhiVelu committed:
fix: validate is_group for parent task

PugazhendhiVelu committed:
test: add test for parent task is_group validation

saadchaudharry committed:
Merge pull request #50299 from saadchaudharry/feature-subcontract-return-rejected

Add support for subcontract return from rejected warehouse

rohitwaghchaure committed:
feat: option to exclude stand-alone returned sales invoices from the Gross Profit report

ravibharathi656 committed:
fix(material request): set default buying price list if not exists

rs-rethik committed:
fix: ignore Department doctype

rehanrehman389 committed:
feat: add asset name column

rehanrehman389 committed:
fix: validate purchase invoice status and resolve related issues

ruthra-kumar committed:
refactor: add paused to select option

rohitwaghchaure committed:
fix: prevent reuse of serial no in manufacture and repack entry

PugazhendhiVelu committed:
fix: hide total row in general ledger report

rehansari26 committed:
fix: set company before creating asset movement to avoid permission error

frappe-pr-bot committed:
fix: sync translations from crowdin (#50281)

PugazhendhiVelu committed:
refactor(task): use get_link_to_form for validation error messages

diptanilsaha committed:
fix: trends report total mismatch with group filters

PatrickDEissler committed:
fix(Timesheet): don't use billing_hours for costing amount calculation (#50392)

rohitwaghchaure committed:
fix: set operating cost based on bom qty causing incorrect operating costing

rohitwaghchaure committed:
feat: Allow Editing of Items and Quantities in Work Order

rohitwaghchaure committed:
fix: removed the validation

diptanilsaha commented on erpnext/buying/doctype/request_for_quotation/test_request_for_quotation.py:
```diff
@@ -76,6 +76,46 @@ def test_make_supplier_quotation(self):
+	def test_make_supplier_quotation_with_taxes(self):
+		"""Test automatic tax addition when supplier quotation is created from RFQ taxes_and_charges are set"""
+
+		# Create a Purchase Taxes and Charges Template for testing
+		tax_template = frappe.new_doc("Purchase Taxes and Charges Template")
+		tax_template.doctype = "Purchase Taxes and Charges Template"
+		tax_template.title = "_Test Purchase Taxes Template for RFQ"
+		tax_template.company = "_Test Company"
+		tax_template.append(
+			"taxes",
+			{
+				"charge_type": "On Net Total",
+				"account_head": "_Test Account Service Tax - _TC",
+				"description": "VAT",
+				"rate": 10,
+			},
+		)
+		tax_template.save()
+
+		rfq = make_request_for_quotation()
+		supplier = rfq.get("suppliers")[0].supplier
+
+		tax_rule = frappe.new_doc("Tax Rule")
+		tax_rule.company = "_Test Company"
+		tax_rule.tax_type = "Purchase"
+		tax_rule.Supplier = supplier
+		tax_rule.purchase_tax_template = tax_template.name
+		tax_rule.save()
+
+		sq = make_supplier_quotation_from_rfq(rfq.name, for_supplier=supplier)
+
+		# Verify that taxes_and_charges is set from get_party_details
+		self.assertEqual(sq.taxes_and_charges, tax_template.name)
+
+		# Verify that taxes are automatically added
+		self.assertGreaterEqual(len(sq.get("taxes")), 1)
+
+		tax_rule.delete()
+		tax_template.delete()
+
```
@coderabbitai resolve.
